src/reference/asciidoc/mqtt.adoc

```diff
@@ -403,3 +403,14 @@ Certain application events are published by the adapters.
 * `MqttSubscribedEvent` - published by the inbound adapter after subscribing to the topics.
 
 These events can be received by an `ApplicationListener<MqttIntegrationEvent>` or with an `@EventListener` method.
+
+To determine the source of an event, use the following; you can check the bean name and/or the connect options (to access the server URIs etc).
+
+====
+[source, java]
+----
+MqttPahoComponent source = event.getSourceAsType();
```

Comment:
Would be great to mention such a type requirement on the `MqttIntegrationEvent` class JavaDocs.

Reply:
I don't see how we can -  the event is not aware of Paho - we have an open issue to migrate to another client which will have a different type.

Reply:
OK. Can we then document it on Paho components about such a specific for events?